File: src/Application Framework/Structures/DataGridSchedulerJobColumn.vb
Public Class DataGridSchedulerJobColumn : Inherits DataGridLabelColumn

#Region "Functions"

    Protected Overloads Overrides Sub Paint(ByVal g As Graphics, ByVal bounds As Rectangle, ByVal source As CurrencyManager, ByVal rowNum As Integer,
                                            ByVal backBrush As Brush, ByVal foreBrush As Brush, ByVal alignToRight As Boolean)

        'Need to get row information from data
        Dim row As DataRow = [source].List.Item(rowNum).row

        If row.Table.Columns.Contains("IsJobLate") Then
            If Entity.Sys.Helper.MakeBooleanValid(row("IsJobLate")) Then
                backBrush = Brushes.Red
            End If
        End If

        If row.Table.Columns.Contains("IsServiceOverDue") Then
            If Entity.Sys.Helper.MakeBooleanValid(row("IsServiceOverDue")) Then
                backBrush = Brushes.Orange
            End If
        End If

        If row.Table.Columns.Contains("Declined") Then
            If Entity.Sys.Helper.MakeBooleanValid(row("Declined")) Then
                backBrush = Brushes.LightGray
            End If
        End If

        MyBase.Paint(g, bounds, source, rowNum, backBrush, foreBrush, alignToRight)
    End Sub

#End Region

End Class

File: src/Application Framework/Structures/DataGridSchedulerJobColumn.cs
using System.Data;
using System.Drawing;
using System.Windows.Forms;

namespace FSM
{
    public class DataGridSchedulerJobColumn : DataGridLabelColumn
    {
        /* TODO ERROR: Skipped RegionDirectiveTrivia */

        protected override void Paint(Graphics g, Rectangle bounds, CurrencyManager source, int rowNum, Brush backBrush, Brush foreBrush, bool alignToRight)
        {
            // Need to get row information from data
            DataRowView row = (DataRowView)source.List[rowNum];
            if (row.DataView.Table.Columns.Contains("IsJobLate"))
            {
                if (Entity.Sys.Helper.MakeBooleanValid(row["IsJobLate"]))
                {
                    backBrush = Brushes.Red;
                }
            }

            if (row.DataView.Table.Columns.Contains("IsServiceOverDue"))
            {
                if (Entity.Sys.Helper.MakeBooleanValid(row["IsServiceOverDue"]))
                {
                    backBrush = Brushes.Orange;
                }
            }

            if (row.DataView.Table.Columns.Contains("Declined"))
            {
                if (Entity.Sys.Helper.MakeBooleanValid(row["Declined"]))
                {
                    backBrush = Brushes.LightGray;
                }
            }

            base.Paint(g, bounds, source, rowNum, backBrush, foreBrush, alignToRight);
        }

        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
    }
}